# MachineLearning UFC fight predictor 

### Andrew Walkowski, Evan Cruse, Jack Kovacs, and Alex Ketcham

NOTE: This if for final development of prediction tools, do brainstorming/exploring in a personal file before adding it here

### 1st import and clean data
Remove Nan values imported in from CSV file for missing fighter records that can not be used in our model

In [3]:
import pandas as pd
import numpy as np
import os

In [4]:
path = os.getcwd()
p = path.removesuffix('ufcPredictionProject')
p= p+'UFCdata.csv'
p

'C:\\Users\\walkowae\\Documents\\GitHub\\UFCdata.csv'

In [8]:
df = pd.read_csv(p)
print('Shape before removing NaN rows: ',df.shape)
df = df.dropna()
print('Shape after: ',df.shape)

Shape before removing NaN rows:  (6012, 144)
Shape after:  (3890, 144)


In [10]:
nunique = df.nunique()
cols_to_drop = nunique[nunique == 1].index
print('Columns where every cell is the same: ', cols_to_drop)
df = df.drop(cols_to_drop, axis=1)
print('New Shape: ',df.shape)

Columns where every cell is the same:  Index([], dtype='object')
New Shape:  (3890, 142)


### 2nd Baseline


In [16]:
y= df.Winner
print('Winner percentages:\n', y.value_counts()/3890)

Winner percentages:
 Red     0.616195
Blue    0.366067
Draw    0.017738
Name: Winner, dtype: float64


In [18]:
baseline = 0.616195**2 + 0.366067**2 +0.017738**2
print(baseline)

0.514015963158


### Baseline = 51.4% (When always guessing Red to win)
### Real World Baseline = 50/50